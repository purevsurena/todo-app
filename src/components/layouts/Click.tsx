import React from "react";
import RNBounceable from "@freakycoder/react-native-bounceable";
import { ClickProps, ClickDefaultProps } from "@/types/layout";

function Click(props: ClickProps): JSX.Element {
  const { onPress, onLongPress, disabled, children, style, hitSlop } = props;

  return (
    <RNBounceable
      testID="clickComponent"
      onPress={onPress}
      onLongPress={onLongPress}
      disabled={disabled}
      hitSlop={{ top: hitSlop, bottom: hitSlop, right: hitSlop, left: hitSlop }}
      style={{ ...style }}
    >
      {children}
    </RNBounceable>
  );
}

Click.defaultProps = ClickDefaultProps;
export default Click;
