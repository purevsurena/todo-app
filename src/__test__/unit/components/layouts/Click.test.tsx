import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Click from "@/components/layouts/Click";
import { Text } from "react-native";

describe("Click Component", () => {
  const defaultProps = {
    onPress: jest.fn(),
    onLongPress: jest.fn(),
    disabled: false,
    children: <Text>Test Children</Text>,
    style: {},
    hitSlop: 10,
  };

  it("renders correctly", () => {
    const { getByText } = render(<Click {...defaultProps} />);
    expect(getByText("Test Children")).toBeTruthy();
  });

  it("calls onPress when pressed", () => {
    const { getByText } = render(<Click {...defaultProps} />);
    fireEvent.press(getByText("Test Children"));
    expect(defaultProps.onPress).toHaveBeenCalled();
  });

  it("calls onLongPress when long pressed", () => {
    const { getByText } = render(<Click {...defaultProps} />);
    fireEvent(getByText("Test Children"), "onLongPress");
    expect(defaultProps.onLongPress).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const { toJSON } = render(<Click {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
