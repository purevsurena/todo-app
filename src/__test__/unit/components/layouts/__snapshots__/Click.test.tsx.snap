// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click Component matches snapshot 1`] = `
<Text>
  Test Children
</Text>
`;
